libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -146,21 +146,25 @@ void AP_AIS::update()
             } else if (_incoming.num == _incoming.total) {
                 // last part of a multi part message
                 uint8_t index = 0;
-                uint8_t msg_parts[_incoming.num - 1];
+
+                // We have the last part, need to find preceding fragments
+                const uint8_t parts = _incoming.num - 1;
```

Comment:
There appears to be an integer underflow problem here; `_incoming.num` is coming off the wire, and nothing is checking if it is zero or not.

This is pre-existing.